### PR TITLE
ci: adds autogen node script

### PR DESCRIPTION
**Pull request by grant**
Per the unmarshal autogen plan, this GitHub workflow runs the generator for the Node client and generates a PR for that Node repo if there is a non-0 diff whenever we push an update to the default branch of this repo.

Please treat the versioning question separate from this PR.

### CODE REVIEW

**Review by grayside — APPROVED** [review]
LGTM but perhaps @averikitsch or someone on the GitHub Automation team should be tagged for review?

**Comment by grant** [comment]
> LGTM but perhaps @averikitsch or someone on the GitHub Automation team should be tagged for review?

Sure, Averi can you take a look at this PR and lmk if there are any issues?

File: .github/workflows/client_node.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+name: Run Node Client Generator and Create Pull Request
+on:
+  push:
+    branches:
+      - "master"
+jobs:
+  create_pr:
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Checkout this repo
```
Comment (averikitsch) [inline .github/workflows/client_node.yaml:24]: Do you need to checkout this repo?

File: .github/workflows/client_node.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        if [[ `git status --porcelain` ]]; then
+          git config --global user.email "yoshi-code-bot@google.com"
+          git config --global user.name "Google Bot"
+          git checkout -b "pr_bot-$GITHUB_RUN_ID"
+          git add -A
+          NOW=$(date +"%m-%d-%Y")
+          git commit --message "feat: run generator (${NOW})"
+          git push origin "pr_bot-$GITHUB_RUN_ID"
+          echo "Created commit: $(git rev-parse HEAD)"
+          gh pr create \
```
Comment (averikitsch) [inline .github/workflows/client_node.yaml:57]: nit: there's an action you can use, but this is just as clean

**Review by averikitsch — APPROVED** [review]
Just nits

File: .github/workflows/client_node.yaml (hunk 1 above)
Comment (grant) [reply .github/workflows/client_node.yaml:24]: Yes, this repo contains schema files used for generation.

File: .github/workflows/client_node.yaml (hunk 2 above)
Comment (grant) [reply .github/workflows/client_node.yaml:57]: Thanks. Using the CLI is easier for manual testing, so let's keep as is.

**Pull request merged**